### PR TITLE
[Agent] Agent 6 final layout

### PR DESCRIPTION
**Pull request by l0k0ms**
## Tl;Dr

Same idea as here: https://github.com/DataDog/documentation/pull/2060
Different layout

Agent 6 is coming: https://github.com/DataDog/datadog-agent
This PR updates the `/agent` section of the doc to document this new version.

## ToC

* **Changes**: All big changes
* **New layout**: Description of the new `/agent` section
* **Review**: Process to review this new PR, which team needs to review which section
* **To Do prior to merging**: what's needs to be done before merging this PR

### New concepts

1. **All documentation pages related to the agent are now in the `/agent` section**

2. **Logs/traces/processes collection docs have a direct entry in the main nav** 

3. **All features that are not linked to the agent installation process or any of the basic agent usages are now grouped in an advanced features section**

4. **All tagging documentation is now in its own sub-section in the getting_started section** [Link to preview](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/getting_started/tagging/)

### New layout
=== Basic agent usage section === 
[Link to the preview](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/basic_agent_usage/)

The index now is a summary allowing to dive into the platforms list, with also a summary of configuration file location and links to agent configuration management tools setup.

All basic agent usage page now document:

* Agent v5 and v6 commands
* How to upgrade from agent v5 to v6
* How to downgrade from agent v6 to v5

=== Log collection === 
[Link to the preview](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/log_collection/)
* [Docker log collection] This content used to be located in `/logs/docker`

=== Trace collection === 
[Link to the preview](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/trace_collection/)

This content is not documented yet and this section is empty, waiting for: https://github.com/DataDog/documentation/pull/2032

=== Process collection === 
[Link to the preview](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/process_collection/) This content used to be located in `/graphing/infrastructure`

* [Docker process collection](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/process_collection/docker_process_collection/)
* [Kubernetes process collection](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/process_collection/kubernetes_process_collection/)

===  Advanced features ===

[Link to the preview](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/advanced_features/)

* [Autodiscovery](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/advanced_features/autodiscovery/)
* [Proxy](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/advanced_features/proxy/)
* [Agent check](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/advanced_features/agent_checks/)
* [Prometheus Check](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/advanced_features/prometheus/)
* [Adding a custom python package to the agent](https://docs-staging.datadoghq.com/gus/agent-6-final-layout/agent/advanced_features/custom_python_package/)

## Review 

* [ ] **Croissant** 
* [ ] **Kenafeh** 
* [x] **Ramen** 
* [ ] **Raclette**
* [ ] **Burrito** 

## To Do prior to merging

* [ ] merge https://github.com/DataDog/dogweb/pull/24458

### CODE REVIEW

File: content/agent/basic_agent_usage/amazonlinux.md [hunk 1]
```diff
@@ -2,86 +2,152 @@
...
-```
+**NB**: If `service` is not available on your system, use:
 
-To restart the Agent and reload the configuration files:
-```shell
-sudo /etc/init.d/datadog-agent restart
-```
+* on `upstart`-based systems: `sudo start/stop/restart datadog-agent`
+* on `systemd`-based systems: `sudo systemctl start/stop/restart datadog-agent`
+* on `initctl`-based systems: `sudo initctl start/stop/restart datadog-agent`
```
Comment (olivielpeau) [inline content/agent/basic_agent_usage/amazonlinux.md]: `initctl` is the same thing as upstart (actually, `initctl` is the command to communicate with upstart). `start`, `stop` and `restart` are aliases for respectively`initctl start`, `initctl stop`, `initctl restart`.

So I think we should merge this line with the `upstart` one and reword it: `* on upstart-based systems: sudo initctl start/stop/restart datadog-agent` (I think it's better to be explicit and use initctl)

This applies to all the linux docs pages :)

**Comment by l0k0ms** [comment]
Let's close this PR.. new concepts introduced are too complex to be tacle here.. let's focus on new agent features for now.

File: content/agent/basic_agent_usage/amazonlinux.md (hunk 1 above)
Comment (l0k0ms) [reply content/agent/basic_agent_usage/amazonlinux.md]: 👌
